### PR TITLE
Agios 525.swift3.release

### PR DESCRIPTION
**Pull request by aidenkeating**

### CODE REVIEW

**Comment by aidenkeating** [comment]
@jcesarmobile Could you take a look at this please?

**Comment by aidenkeating** [comment]
@corinnekrych Would you also mind having a look please?

**Comment by corinnekrych** [comment]
@aidenkeating I think we should revisit the Swift3 API (open method) for ex:
https://github.com/aerogear/aerogear-ios-push/blob/master/push-sdk-swift/DeviceRegistration.swift#L102
should be renamed
```
register(clientInfo: ((config: ClientDeviceInformation) -> Void)!,
        success:(() -> Void)!, failure:((NSError) -> Void)!) -> Void
```
to follow Swift3 API design:
https://github.com/corinnekrych/Swift3Tour/blob/master/Swift3Tour.playground/Pages/API%20Design%20Guide.xcplaygroundpage/Contents.swift#L73

@aidenkeating we missed that in this PR:
https://github.com/aerogear/aerogear-ios-push/pull/89

@jcesarmobile do you agree?

File: push-sdk-swift/DeviceRegistration.swift [hunk 1]
```diff
@@ -204,7 +204,7 @@ open class DeviceRegistration: NSObject, URLSessionTaskDelegate {
           We need to 'override' that 'default' behaviour to return the original attempted NSURLRequest
           with the URL parameter updated to point to the new 'Location' header.
     */
-    open func urlSession(_ session: URLSession, task: URLSessionTask, willPerformHTTPRedirection redirectResponse: HTTPURLResponse, newRequest redirectReq: URLRequest, completionHandler: (@escaping (URLRequest?) -> Void)) {
+    open func urlSession(session: URLSession, task: URLSessionTask, willPerformHTTPRedirection redirectResponse: HTTPURLResponse, newRequest redirectReq: URLRequest, completionHandler: (@escaping (URLRequest?) -> Void)) {
```
Comment (corinnekrych) [inline push-sdk-swift/DeviceRegistration.swift]: this one keep _ because it is inherited.

**Comment by corinnekrych** [comment]
in https://github.com/aerogear/aerogear-ios-push/blob/master/push-sdk-swift/PushAnalytics.swift#L37 and https://github.com/aerogear/aerogear-ios-push/blob/master/push-sdk-swift/PushAnalytics.swift#L55
remove the first `_` to declare a label for the first param.

**Comment by aidenkeating** [comment]
@corinnekrych Are the PushAnalytics changes you recommend not made here?

https://github.com/aerogear/aerogear-ios-push/pull/91/files#diff-07bdc4a2fe51d48489af20c79e9379d3L37

**Comment by corinnekrych** [comment]
@aidenkeating yes! all good ignore that last commnet

**Comment by corinnekrych** [comment]
+1 LGTM
ready for a release

**Pull request merged**